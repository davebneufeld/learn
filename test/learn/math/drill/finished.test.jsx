const constants = require('../../../../src/learn/common/constants');
const React = require('react');
const Finished = require('../../../../src/learn/math/drill/finished');
const renderer = require('react-test-renderer');
const MuiThemeProvider = require('material-ui/styles/MuiThemeProvider').default;
const getMuiTheme = require('material-ui/styles/getMuiTheme').default;
const lightBaseTheme = require('material-ui/styles/baseThemes/lightBaseTheme').default;

const {
  RECORD_NEW,
  // RECORD_EQUAL,
  // RECORD_MISS,
  // RECORD_NOT_EXIST,
} = constants;
const muiTheme = getMuiTheme(lightBaseTheme);

describe('Finished', () => {
  test('should render when there is a single incorrect result', () => {
    const previousResults = [{
      actuals: [5],
      id: 1,
      task: [5, 5, 0, 10],
      timeTaken: 2.2,
    }];
    const resultInfo = {
      text: 'Result Info Text',
      newRecordInfo: RECORD_NEW,
    };

    const component = renderer.create(
      <MuiThemeProvider muiTheme={muiTheme}>
        <Finished
          levelIndex={0}
          opIndexes={[0]}
          previousResults={previousResults}
          resultInfo={resultInfo}
          timeAllowed={600}
          timeLeft={100}
          totalProblems={20}
        />
      </MuiThemeProvider>);

    const finishedBadge = component.toJSON();
    expect(finishedBadge).toMatchSnapshot();
  });

  test('should render when there is a single correct result', () => {
    const previousResults = [{
      actuals: [10],
      id: 1,
      task: [5, 5, 0, 10],
      timeTaken: 2.2,
    }];
    const resultInfo = {
      text: 'Result Info Text',
      newRecordInfo: RECORD_NEW,
    };

    const component = renderer.create(
      <MuiThemeProvider muiTheme={muiTheme}>
        <Finished
          levelIndex={0}
          opIndexes={[0]}
          previousResults={previousResults}
          resultInfo={resultInfo}
          timeAllowed={600}
          timeLeft={100}
          totalProblems={20}
        />
      </MuiThemeProvider>);

    const finishedBadge = component.toJSON();
    expect(finishedBadge).toMatchSnapshot();
  });

  test('should render when there is a mix of correct and incorrect results', () => {
    const previousResults = [{
      actuals: [5], // single incorrect
      id: 1,
      task: [5, 5, 0, 10],
      timeTaken: 2.2,
    }, {
      actuals: [10], // single correct
      id: 2,
      task: [5, 5, 0, 10],
      timeTaken: 2.2,
    }, {
      actuals: [10, 5, 4, 3], // correct with some wrong
      id: 3,
      task: [5, 5, 0, 10],
      timeTaken: 2.2,
    }];
    const resultInfo = {
      text: 'Result Info Text',
      newRecordInfo: RECORD_NEW,
    };

    const component = renderer.create(
      <MuiThemeProvider muiTheme={muiTheme}>
        <Finished
          levelIndex={0}
          opIndexes={[0]}
          previousResults={previousResults}
          resultInfo={resultInfo}
          timeAllowed={600}
          timeLeft={100}
          totalProblems={20}
        />
      </MuiThemeProvider>);

    const finishedBadge = component.toJSON();
    expect(finishedBadge).toMatchSnapshot();
  });
});
