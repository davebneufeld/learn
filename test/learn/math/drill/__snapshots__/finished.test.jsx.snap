// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Finished should be rendered a new record 1`] = `
<div>
  <h1>
    Finished
  </h1>
  <div
    style={
      Object {
        "backgroundColor": "#8bc78b",
        "borderRadius": "5px",
        "color": "black",
        "marginBottom": "10px",
        "padding": "20px",
      }
    }
  >
    Result Info Text
  </div>
  <div
    style={Object {}}
  >
    You need 10 or more correct answers to earn a badge. On this 
test you got 0 answer(s) correct.
  </div>
  <div>
    You had 100 seconds left out of the 600 seconds allowed.
  </div>
  <div>
    You correctly answered 0 of the 20 problems.
  </div>
  <h4>
    The problem that took you the longest to answer
  </h4>
  <div>
    <div
      style={
        Object {
          "color": "red",
          "lineHeight": "35px",
        }
      }
    >
      <div
        className={undefined}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "transparent",
            "borderRadius": "50%",
            "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
         0 3px 10px rgba(0, 0, 0, 0.23)",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "1.5em",
            "margin": "5px",
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        <button
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "backgroundColor": "#c0c0c0",
              "border": 10,
              "borderRadius": "50%",
              "boxSizing": "border-box",
              "color": "white",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "inherit",
              "fontWeight": "inherit",
              "height": 40,
              "margin": 0,
              "muiPrepared": true,
              "outline": "none",
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "textAlign": "center",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": "bottom",
              "width": 40,
              "zIndex": 1,
            }
          }
          tabIndex={0}
          title="+ Silver"
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <div
              style={
                Object {
                  "muiPrepared": true,
                  "top": 0,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                }
              }
            >
              +
            </div>
          </div>
        </button>
      </div>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        5
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        +
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        5
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        =
      </span>
      <span
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 1,
            "marginLeft": 20,
            "padding": 3,
          }
        }
      >
        5
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        ✗
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        2.2 seconds
      </span>
    </div>
  </div>
  <h4>
    The problem you answered the fastest
  </h4>
  <div>
    <div
      style={
        Object {
          "color": "red",
          "lineHeight": "35px",
        }
      }
    >
      <div
        className={undefined}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "transparent",
            "borderRadius": "50%",
            "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
         0 3px 10px rgba(0, 0, 0, 0.23)",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "1.5em",
            "margin": "5px",
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        <button
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "backgroundColor": "#c0c0c0",
              "border": 10,
              "borderRadius": "50%",
              "boxSizing": "border-box",
              "color": "white",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "inherit",
              "fontWeight": "inherit",
              "height": 40,
              "margin": 0,
              "muiPrepared": true,
              "outline": "none",
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "textAlign": "center",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": "bottom",
              "width": 40,
              "zIndex": 1,
            }
          }
          tabIndex={0}
          title="+ Silver"
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <div
              style={
                Object {
                  "muiPrepared": true,
                  "top": 0,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                }
              }
            >
              +
            </div>
          </div>
        </button>
      </div>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        5
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        +
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        5
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        =
      </span>
      <span
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 1,
            "marginLeft": 20,
            "padding": 3,
          }
        }
      >
        5
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        ✗
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        2.2 seconds
      </span>
    </div>
  </div>
  <div>
    <h4>
      The ones that you got wrong
    </h4>
    <div>
      <div
        style={
          Object {
            "color": "red",
            "lineHeight": "35px",
          }
        }
      >
        <div
          className={undefined}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "backgroundColor": "transparent",
              "borderRadius": "50%",
              "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
         0 3px 10px rgba(0, 0, 0, 0.23)",
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "1.5em",
              "margin": "5px",
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          <button
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "backgroundColor": "#c0c0c0",
                "border": 10,
                "borderRadius": "50%",
                "boxSizing": "border-box",
                "color": "white",
                "cursor": "pointer",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": "inherit",
                "fontWeight": "inherit",
                "height": 40,
                "margin": 0,
                "muiPrepared": true,
                "outline": "none",
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "textAlign": "center",
                "textDecoration": "none",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "verticalAlign": "bottom",
                "width": 40,
                "zIndex": 1,
              }
            }
            tabIndex={0}
            title="+ Silver"
            type="button"
          >
            <div
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
            >
              <div
                style={
                  Object {
                    "muiPrepared": true,
                    "top": 0,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  }
                }
              >
                +
              </div>
            </div>
          </button>
        </div>
        <span
          style={
            Object {
              "paddingLeft": 20,
            }
          }
        >
          5
        </span>
        <span
          style={
            Object {
              "paddingLeft": 20,
            }
          }
        >
          +
        </span>
        <span
          style={
            Object {
              "paddingLeft": 20,
            }
          }
        >
          5
        </span>
        <span
          style={
            Object {
              "paddingLeft": 20,
            }
          }
        >
          =
        </span>
        <span
          style={
            Object {
              "borderStyle": "solid",
              "borderWidth": 1,
              "marginLeft": 20,
              "padding": 3,
            }
          }
        >
          5
        </span>
        <span
          style={
            Object {
              "paddingLeft": 20,
            }
          }
        >
          ✗
        </span>
        <span
          style={
            Object {
              "paddingLeft": 20,
            }
          }
        >
          2.2 seconds
        </span>
      </div>
    </div>
  </div>
  <h4>
    How you did (sorted by slowest)
  </h4>
  <div>
    <div
      style={
        Object {
          "color": "red",
          "lineHeight": "35px",
        }
      }
    >
      <div
        className={undefined}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "transparent",
            "borderRadius": "50%",
            "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
         0 3px 10px rgba(0, 0, 0, 0.23)",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "1.5em",
            "margin": "5px",
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        <button
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "backgroundColor": "#c0c0c0",
              "border": 10,
              "borderRadius": "50%",
              "boxSizing": "border-box",
              "color": "white",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "inherit",
              "fontWeight": "inherit",
              "height": 40,
              "margin": 0,
              "muiPrepared": true,
              "outline": "none",
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "textAlign": "center",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": "bottom",
              "width": 40,
              "zIndex": 1,
            }
          }
          tabIndex={0}
          title="+ Silver"
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <div
              style={
                Object {
                  "muiPrepared": true,
                  "top": 0,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                }
              }
            >
              +
            </div>
          </div>
        </button>
      </div>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        5
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        +
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        5
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        =
      </span>
      <span
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 1,
            "marginLeft": 20,
            "padding": 3,
          }
        }
      >
        5
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        ✗
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        2.2 seconds
      </span>
    </div>
  </div>
</div>
`;
